app/Http/Controllers/UserGitHubUsernameController.php

```diff
@@ -20,51 +19,31 @@
      */
     public function index(): RedirectResponse
     {
-        return type(Socialite::driver('github')->redirect())->as(RedirectResponse::class);
+        /** @var GithubProvider $driver */
+        $driver = Socialite::driver('github');
+
+        return $driver->redirectUrl(route('profile.connect.github.update'))->redirect();
     }
 
     /**
      * Handles the GitHub connection update.
      */
-    public function update(Request $request): RedirectResponse
+    public function update(Request $request, GitHub $github): RedirectResponse
     {
         $githubUser = Socialite::driver('github')->user();
 
         $user = type($request->user())->as(User::class);
 
-        try {
-            $validated = Validator::validate([
-                'github_username' => $githubUser->getNickname(),
-            ], [
-                'github_username' => ['required', 'string', 'max:255', 'unique:users,github_username'],
-            ], [
-                'github_username.unique' => 'This GitHub username is already connected to another account.',
-            ]);
-        } catch (ValidationException $e) {
-            if ($githubUser->getNickname() === $user->github_username) {
-                session()->flash('flash-message', 'The same GitHub account has been connected.');
-
-                return to_route('profile.edit');
-            }
-
-            return to_route('profile.edit')->withErrors($e->errors(), 'verified');
-        }
+        $githubUsername = $githubUser->getNickname();
 
-        $user->update($validated);
+        $errors = $github->linkGitHubUser($githubUsername, $user, $request);
```

Comment:
We moved all this code to a service so that it can be reused in the other two cases where we want to link a user to their GitHub account